frontend/src/employee-frontend/App.tsx

```diff
@@ -416,7 +423,7 @@ function RedirectToMainPage() {
   } else if (hasRole(roles, 'DIRECTOR')) {
     return <Redirect to={'/reports'} />
   } else {
-    return <Redirect to={'/search'} />
+    return <Redirect to={'/welcome'} />
```

Comment:
Now it forwards some of the roles to welcome page because this if-else chain is not exhaustive. 

You need to add something like
`if(roles.length === 0) return <Redirect to={'/welcome'} />`
on line 415 instead